lib/xdp_loader.h

```diff
@@ -0,0 +1,31 @@
+#ifndef __XDP_LOADER_H_
+#define __XDP_LOADER_H_
+
+using namespace std;
+
+class XdpLoader {
+    public:
+        XdpLoader(): handle(nullptr) { }
+        ~XdpLoader();
+        int LoadFrmFile(const string filepath, const int ifindex);
+        int LoadFrmBuffer(const string_view buffer, const char *name);
+    private:
+        void *handle;
```

Comment:
XdpLoader is no longer needed, however in the future you should prefer to initialize member variables in-line instead of in the constructor. If a variable is initialized to a constant in the constructor, the addition of a new constructor may not duplicate the initialization.